.github/workflows/test.yml

```diff
@@ -0,0 +1,24 @@
+name: Test
```

Comment:
Let's make this `test-commit.yml` and question:

We want this to NOT run on a push to main (since same tests are run by `build-and-push-to-main.yml`. Does it already work like that or needs some kind of `ignore`?

Reply:
Currently it runs when a pull request is made.